### PR TITLE
lib/external: Add parson library for JSON

### PR DESCRIPTION
**Pull request by cwhite911**
For more details please view discussion at https://github.com/OSGeo/grass/discussions/3019

Issue: https://github.com/OSGeo/grass/issues/3020

### CODE REVIEW

**Comment by cwhite911** [comment]
@marisn not sure what I'm missing here. Will you please take a look if I forgot to add something anywhere so this will build.

**Comment by neteler** [comment]
`PARSON:parson` needed also to be registered in `include/Make/Grass.make`. Fixed in 2c492e2.

**Comment by cwhite911** [comment]
Thanks @neteler! I will write up documentations for this process. Would you rather it be it's own wiki or added to the GRASS Programmer's manual under "Further libraries" after "Miscellaneous Libraries"? I could also add a README.md to the lib/external folder to document the process.

**Review by marisn — APPROVED** [review]
Seems good to me.

**Review by wenzeslaus — APPROVED** [review]

File: grasslib.dox [hunk 1]
```diff
@@ -131,6 +131,22 @@ href="https://grass.osgeo.org">https://grass.osgeo.org</a>
...
+ - <b>add:</b> lib/external/<b>newlib</b>
+ - <b>add:</b> lib/external/<b>newlib</b>/*.c
+ - <b>add:</b> lib/external/<b>newlib</b>/*.h
+ - <b>add:</b> lib/external/<b>newlib</b>/LICENSE
+ - <b>add:</b> lib/external/<b>newlib</b>/Makefile
+ - <b>update:</b> lib/external/Makefile
+ - <b>update:</b> lib/external/README.license
+ - <b>update:</b> lib/README.md
+ - <b>update:</b> include/Make/Install.make
+ - <b>update:</b> include/Make/GRASS.make
```
Comment (wenzeslaus) [inline grasslib.dox:147]:
```suggestion
 - <b>update:</b> include/Make/Grass.make
```

**Review by wenzeslaus — COMMENTED** [review]
I don't know why this section should be here rather than in the readme, but it is not related to this PR and I don't want to delay this any longer.

**Comment by landam** [comment]
@wenzeslaus Please consider merging this PR.

File: lib/external/README.md [hunk 2]
```diff
@@ -0,0 +1,120 @@
+# How to add a new external library
+
+The follow procedure will walk you through adding a new third-party
+library to GRASS GIS.
+
+## License
+
+Before you begin make sure the library you want to include is complient with
+GRASS GIS's license.
```
Comment (neteler) [inline lib/external/README.md:9]:
```suggestion
Before you start, make sure that the library you want to include is in
compliance with the GRASS GIS licence.
```

File: lib/external/README.md [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+* MPL 2.0
+* BSD-3-Clause
+* Public Domain
+
+## Procedure
+
+* Create a directory that will contain the new library's files
+in `lib/external/<new-library>`
+
+* Add the new libraries header, c, license, and README to the newly
```
Comment (neteler) [inline lib/external/README.md:25]: "c" is unclear to me -> "copyright"?

File: lib/external/README.md [hunk 4]
```diff
@@ -0,0 +1,120 @@
+# How to add a new external library
+
+The follow procedure will walk you through adding a new third-party
```
Comment (petrasovaa) [inline lib/external/README.md:3]:
```suggestion
The following procedure will walk you through adding a new third-party
```

File: lib/external/README.md (hunk 3 above)
Comment (petrasovaa) [reply lib/external/README.md:25]:
```suggestion
* Add the new libraries header and source files, license, and README to the newly
```

**Pull request merged**